stream.go

```diff
@@ -140,6 +144,198 @@ func (r *Client) StreamPrediction(ctx context.Context, prediction *Prediction) (
 	return sseChan, errChan
 }
 
+func (r *Client) StreamPredictionFiles(ctx context.Context, prediction *Prediction) (<-chan streaming.File, error) {
+	url := prediction.URLs["stream"]
+	if url == "" {
+		return nil, errors.New("streaming not supported or not enabled for this prediction")
+	}
+
+	ch := make(chan streaming.File)
+
+	go r.streamFilesTo(ctx, ch, url, "")
+	return ch, nil
+}
+
+func (r *Client) StreamPredictionText(ctx context.Context, prediction *Prediction) (io.Reader, error) {
+	url := prediction.URLs["stream"]
+	if url == "" {
+		return nil, errors.New("streaming not supported or not enabled for this prediction")
+	}
+
+	reader, writer := io.Pipe()
+
+	go r.streamTextTo(ctx, writer, url, "")
+	return reader, nil
+}
+
+func (r *Client) streamTextTo(ctx context.Context, writer *io.PipeWriter, url string, lastEventID string) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		writer.CloseWithError(fmt.Errorf("failed to create request: %w", err))
+		return
+	}
+	req.Header.Set("Accept", "text/event-stream")
+	req.Header.Set("Cache-Control", "no-cache")
+	req.Header.Set("Connection", "keep-alive")
```

Comment:
As mentioned, I'm not really sure either `Cache-Control` or `Connection` are doing anything useful here, so I'd be tempted to drop them.